src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -278,220 +263,229 @@ function createEmscriptenModuleInstance(resourceLoader: WebAssemblyResourceLoade
 
-      if (resourceLoader.startOptions.applicationCulture) {
-        // If a culture is specified via start options use that to initialize the Emscripten \  .NET culture.
-        MONO.mono_wasm_setenv('LANG', `${resourceLoader.startOptions.applicationCulture}.UTF-8`);
+      try {
+        // Wait for the data to be loaded and verified
+        const dataBuffer = await dependency.response.then(r => r.arrayBuffer());
+
+        // Load it into the Mono runtime
+        const data = new Uint8Array(dataBuffer);
+        const heapAddress = Module._malloc(data.length);
+        const heapMemory = new Uint8Array(Module.HEAPU8.buffer, <any>heapAddress, data.length);
+        heapMemory.set(data);
+        MONO.mono_wasm_add_assembly(loadAsName, heapAddress, data.length);
+        MONO.loaded_files.push(toAbsoluteUrl(dependency.url));
+      } catch (errorInfo) {
+        onError(errorInfo);
+        return;
       }
-    }
-    let timeZone = 'UTC';
-    try {
-      timeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;
-    // eslint-disable-next-line no-empty
-    } catch { }
-    MONO.mono_wasm_setenv('TZ', timeZone || 'UTC');
-    if (resourceLoader.bootConfig.modifiableAssemblies) {
-      // Configure the app to enable hot reload in Development.
-      MONO.mono_wasm_setenv('DOTNET_MODIFIABLE_ASSEMBLIES', resourceLoader.bootConfig.modifiableAssemblies);
-    }
 
-    if (resourceLoader.bootConfig.aspnetCoreBrowserTools) {
-      // See https://github.com/dotnet/aspnetcore/issues/37357#issuecomment-941237000
-      MONO.mono_wasm_setenv('__ASPNETCORE_BROWSER_TOOLS', resourceLoader.bootConfig.aspnetCoreBrowserTools);
+      Module.removeRunDependency(runDependencyId);
     }
 
-    const load_runtime = cwrap('mono_wasm_load_runtime', null, ['string', 'number']);
-    // -1 enables debugging with logging disabled. 0 disables debugging entirely.
-    load_runtime(appBinDirName, hasDebuggingEnabled() ? -1 : 0);
-    MONO.mono_wasm_runtime_ready();
-    attachInteropInvoker();
-    onReady();
-  });
-
-  return module;
-
-  async function addResourceAsAssembly(dependency: LoadingResource, loadAsName: string) {
-    const runDependencyId = `blazor:${dependency.name}`;
-    addRunDependency(runDependencyId);
-
-    try {
-      // Wait for the data to be loaded and verified
-      const dataBuffer = await dependency.response.then(r => r.arrayBuffer());
-
-      // Load it into the Mono runtime
-      const data = new Uint8Array(dataBuffer);
-      const heapAddress = Module._malloc(data.length);
-      const heapMemory = new Uint8Array(Module.HEAPU8.buffer, heapAddress, data.length);
-      heapMemory.set(data);
-      mono_wasm_add_assembly(loadAsName, heapAddress, data.length);
-      MONO.loaded_files.push(toAbsoluteUrl(dependency.url));
-    } catch (errorInfo) {
-      onError(errorInfo);
-      return;
-    }
+    const dotnetModuleConfig: DotnetModuleConfig = {
+      ...moduleConfig,
+      disableDotnet6Compatibility: false,
```

Comment:
We are still exporting methods and objects to `window` or `globalThis` when this is `false`. 
That is bad practice, but backward compatible.
You could consider if your users expect those symbols to be there in Net7.0 or if you are OK breaking those undocumented assumptions.

Reply:
Agreed on the goals. We have more work to do if we want Blazor to be loadable side-by-side, as (eg) the JS interop APIs can be called from global scope without saying which .NET instance you're trying to call. We'd need to make breaking APIs changes if we want side-by-side by default.